Sources/SwiftFormatRules/BlankLineBetweenMembers.swift

```diff
@@ -18,122 +18,62 @@ import SwiftSyntax
+        && memberTrivia.numberOfLeadingNewlines == 1  // Only one newline => no blank line
+      {
+        let correctTrivia = Trivia.newlines(1) + memberTrivia
+        memberToAdd = replaceTrivia(
+          on: memberToAdd, token: memberToAdd.firstToken!,
           leadingTrivia: correctTrivia
         ) as! MemberDeclListItemSyntax
-        
-        diagnose(.addBlankLine, on: currentMember)
-        hasValidNumOfBlankLines = false
-        membersList.append(newMember)
-      }
-      else {
-        membersList.append(member)
-      }
-    }
-    
-    return hasValidNumOfBlankLines ? node :
-      node.withMembers(SyntaxFactory.makeMemberDeclList(membersList))
-  }
-  
-  /// Indicates if the given trivia has more than
-  /// the maximum number of blank lines.
-  func exceedsMaxBlankLines(_ trivia: Trivia) -> Bool {
-    let maxBlankLines = context.configuration.maximumBlankLines
 
-    for piece in trivia {
-      if case .newlines(let num) = piece,
-        num - 1 > maxBlankLines {
-        return true
+        diagnose(.addBlankLine, on: memberToAdd)
       }
-    }
-    return false
-  }
-  
-  /// Returns the given trivia without any set of consecutive blank lines
-  /// that exceeds the maximumBlankLines.
-  func removeExtraBlankLines(_ trivia: Trivia, _ member: MemberDeclListItemSyntax) -> Trivia {
-    let maxBlankLines = context.configuration.maximumBlankLines
-    var pieces = [TriviaPiece]()
-    
-    // Iterates through the trivia, verifying that the number of blank
-    // lines in the file do not exceed the maximumBlankLines. If it does
-    // a lint error is raised.
-    for piece in trivia {
-      if case .newlines(let num) = piece,
-         num - 1 > maxBlankLines {
-        pieces.append(.newlines(maxBlankLines + 1))
-        diagnose(.removeBlankLines(count: num - maxBlankLines), on: member)
-      }
-      else {
-        pieces.append(piece)
-      }
-    }
-    return Trivia(pieces: pieces)
-  }
-
-  /// Indicates if a declaration has to be ignored by checking if it's
-  /// a single line and if the format is configured to ignore single lines.
-  func ignoreItem(item: MemberDeclListItemSyntax) -> Bool {
-    guard let firstToken = item.firstToken else { return false }
-    guard let lastToken = item.lastToken else { return false }
 
-    let firstTokenLocation = context.sourceLocationConverter.location(
-      for: firstToken.positionAfterSkippingLeadingTrivia)
-    let lastTokenLocation = context.sourceLocationConverter.location(
-      for: lastToken.positionAfterSkippingLeadingTrivia)
-    let isSingleLine = firstTokenLocation.line == lastTokenLocation.line
-
-    let ignoreLine = context.configuration.blankLineBetweenMembers
-      .ignoreSingleLineProperties
+      membersList.append(memberToAdd)
+      previousMemberWasSingleLine = memberIsSingleLine
+    }
 
-    return isSingleLine && ignoreLine
+    return node.withMembers(SyntaxFactory.makeMemberDeclList(membersList))
```

Comment:
Is is safe to use `SyntaxFactory` here because of the incomplete AST problem?

Reply:
Unfortunately there isn't a `with*` method for collections (i.e., "give me the same collection but with all the elements replaced with different ones). The best we can do is remove them all and append new ones, which feels horribly inefficient... so I need to investigate what the right thing to do here is.

But `SyntaxFactory` was used previously here, so it's not making things _worse_ than they were already.